File: leetcode88.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        leftPointer = 0
        rightPointer = 0
        holder = []
        if len(nums2) == 0:
            return nums1
        else:
            while rightPointer < len(nums2):
                if leftPointer == m:
                    holder.extend(nums2[rightPointer:])
                    for x in range(m+n):
                        nums1[x] = holder[x]  
                    return 
                elif nums1[leftPointer] < nums2[rightPointer]:
                    holder.append(nums1[leftPointer])
                    leftPointer += 1
                else:
                    holder.append(nums2[rightPointer])
                    rightPointer += 1
            
            cleanup = len(nums1)-1
            while leftPointer < cleanup:
                if leftPointer == m:
                    for x in range(m+n):
                        nums1[x] = holder[x]  
                    return
                else:
                    holder.append(nums1[leftPointer])
                    leftPointer += 1
            for x in range(m+n):
                nums1[x] = holder[x]

File: javaStuff/leetcode88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] result = new int[m+n];
        int pointer1 = 0;
        int pointer2 = 0;
        int counter = 0;
        while(pointer1 < m && pointer2 < n){
            if(nums1[pointer1] < nums2[pointer2]){
                result[counter] = nums1[pointer1];
                counter++;
                pointer1++;
            }
            else{
                result[counter] = nums2[pointer2];
                counter++;
                pointer2++;
            }
        }
        while(pointer1 < m){
            result[counter] = nums1[pointer1];
            counter++;
            pointer1++;
        }
        while(pointer2 < n){
            result[counter] = nums2[pointer2];
            counter++;
            pointer2++;
        }
        for(int i = 0; i < nums1.length; i++){
            nums1[i] = result[i];
        }
    }
}
